data/shared/src/main/scala/org/ergoplatform/validation/ValidationRules.scala

```diff
@@ -162,6 +164,47 @@ object ValidationRules {
     override protected def settings: SigmaValidationSettings = currentSettings
   }
 
+  object CheckV6Type extends ValidationRule(1019,
+    "Check the type has the declared method.") {
+    override protected lazy val settings: SigmaValidationSettings = currentSettings
+
+    final def apply[T](v: EvaluatedValue[_]): Unit = {
+      checkRule()
+
+      def v6TypeCheck(tpe: SType) = {
+        if (tpe.isOption || tpe.typeCode == SHeader.typeCode || tpe.typeCode == SUnsignedBigInt.typeCode) {
+          throwValidationException(
+            SerializerException(s"V6 type used in register or context var extension: $tpe"),
+            Array[Any](tpe))
+        }
+      }
+
+      def step(s: SType): Unit = {
```

Comment:
Doesn't seem to be sufficient for Coll type that's nested in another type like Tuple or Coll

```scala
    val b7 = new ErgoBoxCandidate(1L, trueProp, 1,
      additionalRegisters = Map(R4 -> ConcreteCollection(Seq(ConcreteCollection(Seq(UnsignedBigIntConstant(new BigInteger("1"))), SUnsignedBigInt)), (SCollection(SUnsignedBigInt)))))
    VersionContext.withVersions(3, 3) {
      val bs = ErgoBoxCandidate.serializer.toBytes(b7)
      a[sigma.validation.ValidationException] should be thrownBy ErgoBoxCandidate.serializer.fromBytes(bs)
    }
```

Reply:
oh right, fixed in https://github.com/ergoplatform/sigmastate-interpreter/pull/1047/commits/845c3e02023e2c02a0d10f4b05c2b4a478690caa